crates/subspace-farmer/src/ws_rpc_server.rs

```diff
@@ -430,17 +426,23 @@ impl RpcServerImpl {
     /// Read and decode the whole piece
     async fn read_and_decode_piece(&self, piece_index: PieceIndex) -> Result<Piece, Error> {
         let piece_fut = tokio::task::spawn_blocking({
-            let plot = self.plot.clone();
+            let plots = self.plots.clone();
 
-            move || plot.read(piece_index)
+            move || {
+                plots.iter().find_map(|plot| {
```

Comment:
We need TODO about more efficient retrieval based on closest node ID as well, here and below.